Geometry/CaloGeometry/interface/CaloCellGeometry.h

```diff
@@ -76,13 +77,16 @@ class CaloCellGeometry
   virtual ~CaloCellGeometry() ;
       
   /// Returns the corner points of this cell's volume.
-  CornersVec const & getCorners() const { assert(not m_corners.uninitialized()); return m_corners; }
-  RepCorners const & getCornersREP() const {  return m_repCorners;}
-
+  const CornersVec& getCorners() const { assert(not m_corners.uninitialized()); return m_corners; }
+  const RepCorners& getCornersREP() const {  return m_repCorners;}
+  virtual const std::vector<GlobalPoint> getCorners(const DetId&) const;
```

Comment:
if its const, just return a reference to m_corners?

Reply:
It can be returned by reference for ECAL, HCAL but not for HGCal. It is computed for each DetId. May be remove const

Reply:
I would leave const instead of adding internal structure to be safe, even if some performance penalty does exist.